Update golang.md (#3479)
TykTechnologies/tyk-docs: tyk-docs/content/plugins/supported-languages/golang.md

## Plugin development flow

Initialising the gateway has slightly changed over time. 
In this section, you will find instructions for initializing the gateway for different versions of Tyk Gateway, before v5.1 and 5.1 onwards. Please ensure that you follow the correct section based on your Gateway version.

### Initialise plugin for Gateway v5.1 and above (v5.1+)

**If you are using Gateway version 5.1 or higher, please follow the steps outlined in this section. These instructions are tailored to the latest Gateway software.**

> In Gateway version 5.1, the gateway and plugins transitioned to a [Go modules](https://go.dev/ref/mod#introduction) build and don't use [govendor](https://pkg.go.dev/github.com/kardianos/govendor) anymore. To create a full workspace follow the guide below. To find out more about Go workspaces, visit the [official documentation](https://go.dev/ref/mod#workspaces).

Reviewer: ```suggestion
In Gateway version 5.1, the Gateway and plugins transitioned to a [Go modules](https://go.dev/ref/mod#introduction) build and don't use [Go vendor](https://pkg.go.dev/github.com/kardianos/govendor) anymore. To create a full workspace follow the guide below. To find out more about Go workspaces, visit the [official documentation](https://go.dev/ref/mod#workspaces).
```

You need two checkouts:

1. Your plugin sources (using `tyk_plugin`)
2. Tyk gateway source for the release (using `tyk`)

In the parent folder, run the following commands to create a go workspace:

```console
go work init ./tyk
go work use ./tyk_plugin
```

To build compatible plugins, the gateway dependency that `tyk_plugin` uses needs to match the checkout in the tyk folder. To get the commit hash, go into `tyk` and run `git rev-parse HEAD`. To synchronize the gateway dependency go into `tyk_plugin` and run `go get github.com/TykTechnologies/tyk@<commit-hash>`.

To gain a deeper understanding of `go work`, you can explore the [Go multi-module workspaces tutorial](https://go.dev/doc/tutorial/workspaces) on the official Go website.

### Initialise plugin for Gateway versions earlier than 5.1

> If you are using Tyk Gateway version that is older than 5.1, please use this section. The steps provided here are specific to Tyk Gateway versions below c5.1.

Reviewer: ```suggestion
If you are using Tyk Gateway version that is older than 5.1, please use this section. The steps provided here are specific to Tyk Gateway versions below c5.1.
```

> If you are using Tyk Gateway version that is older than 5.1, please use this section. The steps provided here are specific to Tyk Gateway versions below c5.1.

Reviewer: ```suggestion
If you are using a Tyk Gateway version that is older than 5.1, please use this section. The steps provided here are specific to Tyk Gateway versions below v5.1.
```
